lib/src/command/deps.dart

```diff
@@ -231,4 +263,61 @@ class DepsCommand extends PubCommand {
         'was generated, please run "pub get" again.');
     return null;
   }
+
+  /// Outputs all executables reachable from [entrypoint].
+  void _outputExecutables() {
+    var packages = []
+      ..add(entrypoint.root)
+      ..addAll((_includeDev
+              ? entrypoint.root.immediateDependencies
+              : _nonDevDependencies(entrypoint.root))
+          .map((dep) => entrypoint.packageGraph.packages[dep.name]));
+
+    for (var package in packages) {
+      var executables = _getExecutablesFor(package);
+      if (executables.isNotEmpty) {
+        _buffer.writeln(_formatExecutables(package.name, executables.toList()));
+      }
+    }
+  }
+
+  /// Lists all Dart files in the `bin` directory of the [package].
+  ///
+  /// Returns file names without extensions.
+  List<String> _getExecutablesFor(Package package) => package
+      .listFiles(beneath: 'bin', recursive: false)
```

Comment:
Hmm, good question. To me it seems like executableIds should already be doing an equivalent `_isDartExecutable` check, but technically changing it to do that would be a breaking change.

The actual `isEntrypoint` check for instance doesn't actually catch all cases, so technically we could break existing packages so you could no longer run their executable. In reality my guess is it would not break anything, but its possible it would.

cc @kevmoo @munificent what do you think?